### PR TITLE
Add java solution to mooyo mooyo

### PR DESCRIPTION
**Pull request by nathangong**

### CODE REVIEW

File: solutions/usaco-860.mdx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+public class MooyoMooyo {
+	static int n, k;
+	static int[][] grid, region;
+	
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("mooyomooyo");
+
+		n = io.nextInt();
+		k = io.nextInt();
+		grid = new int[n][10];
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:31]: magic numbers are bad practice, could you put it in a final variable like `ROW_WIDTH` or something

File: solutions/usaco-860.mdx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+					if (largeRegions.contains(region[i][j])) {
+						grid[i][j] = 0;
+					}
+				}
+			}
+			
+			// Apply gravity; In other words, for each column, we move all the
+			// zeros to the top
+			// Here's a good source explaining a similar concept:
+			// https://www.geeksforgeeks.org/move-zeroes-end-array/
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:81]: multiline comment?

File: solutions/usaco-860.mdx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+		while (true) {
+			// Stores the current region we're on
+			int r = 1;
+			// Reset region matrix
+			region = new int[n][10];
+
+			HashSet<Integer> largeRegions = new HashSet<>();
+			for (int i = 0; i < n; i++) {
+				for (int j = 0; j < 10; j++) {
+					// Floodfill only if the current square isn't empty
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:51]: multiline comment here

File: solutions/usaco-860.mdx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+		region = new int[n][10];
+
+		for (int i = 0; i < n; i++) {
+			String row = io.next();
+			for (int j = 0; j < 10; j++) {
+				grid[i][j] = Character.getNumericValue(row.charAt(j));
+			}
+		}
+
+		// Begin game loop
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:41]: maybe "loop until the thing ends" or something

File: solutions/usaco-860.mdx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+			}
+		}
+
+		for (int i = 0; i < n; i++) {
+			for (int j = 0; j < 10; j++) {
+				io.print(grid[i][j]);
+			}
+			io.println();
+		}
+
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:102]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
it's pog just some minor things

File: solutions/usaco-860.mdx (hunk 1 above)
Comment (ar492) [reply solutions/usaco-860.mdx:31]: don't think it matters; however, it would be nice if 10 was an appropriately named variable

**Review by ar492 — CHANGES_REQUESTED** [review]

File: solutions/usaco-860.mdx (hunk 1 above)
Comment (nathangong) [reply solutions/usaco-860.mdx]: I just named it m because that's pretty standard for usaco

File: solutions/usaco-860.mdx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+			// Delete squares that are part of a large region
+			for (int i = 0; i < n; i++) {
+				for (int j = 0; j < m; j++) {
+					if (largeRegions.contains(region[i][j])) {
+						grid[i][j] = 0;
+					}
+				}
+			}
+			
+			/* Apply gravity; In other words, for each column, we move all the
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:80]:
```java
/*
 * multiline comments should be
 * styled something like this
 */

**Review by SansPapyrus683 — APPROVED** [review]

**Review by nwatx — APPROVED** [review]
Code looks good

**Review by frodakcin — APPROVED** [review]
Skimmed it; looks ok

**Review by ar492 — APPROVED** [review]
Looks good.

File: solutions/usaco-860.mdx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+[Official Analysis](http://www.usaco.org/current/data/sol_mooyomooyo_silver_dec18.html)
+
+## Implementation
+
+<LanguageSection>
+
+<JavaSection>
+
+**Time Complexity:** $\mathcal{O}(N^2)$
```
Comment (HarryWangATX) [inline solutions/usaco-860.mdx:16]: outside

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: solutions/usaco-860.mdx [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+		}
+
+		// Loop until game ends (when there are no more regions to delete)
+		while (true) {
+			// Stores the current region we're on
+			int r = 1;
+			// Reset region matrix
+			region = new int[n][m];
+
+			HashSet<Integer> largeRegions = new HashSet<>();
```
Comment (HarryWangATX) [inline solutions/usaco-860.mdx:49]: use polymorphism pls.

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

**Review by HarryWangATX — APPROVED** [review]

**Pull request merged**